lib/containers/utils.pm

```diff
@@ -154,23 +154,31 @@ sub basic_container_tests {
 # Setup environment
 sub container_set_up {
     my ($dir, $file, $base) = @_;
-    die "You must define the directory!"  unless $dir;
-    die "You must define the Dockerfile!" unless $file;
-    my $basename_expected = script_run("grep baseimage_var $dir/BuildTest/$file");
-    die "Base image name is required for $file" if !$basename_expected && $base;
+    if (!defined($dir) || !!($dir =~ /^ *$/)) {
+        die 'You must define build directory in SUT!';
+    }
+    elsif (!defined($file) || !!($file =~ /^ *$/)) {
+        die 'You must define a template Dockerfile from data/containers/ directory!';
+    }
+    else {
+        assert_script_run "mkdir -p $dir/BuildTest/templates";
+    }
 
-    record_info "Dockerfile: $file";
-    assert_script_run("mkdir -p $dir/BuildTest");
+    record_info('Dockerfile Path', $file);
     assert_script_run "curl -f -v " . data_url('containers/app.py') . " > $dir/BuildTest/app.py";
-    record_info('app.py', script_output("cat $dir/BuildTest/app.py"));
+    record_info('app.py', 'Content of the app is available on serial console');
+    assert_script_run "cat $dir/BuildTest/app.py | tee -a /dev/$serialdev";
```

Comment:
do we need to `tee` to a file? we can just run the command and the output will be available anyway in `serial_terminal.txt`

Reply:
Many backends does not support virtio console.

Reply:
I find it misleading to use `assert_script_run` together with `tee` . your assertion will always pass no matter what you are doing before it . please switch to `script_run`

Reply:
> I find it misleading to use `assert_script_run` together with `tee` . your assertion will always pass no matter what you are doing before it . please switch to `script_run`

I assume that pipefail is set.

```
❯ set -o pipefail; cat asd | tee test; echo $?
cat: asd: No such file or directory
1
```

Reply:
ok make sense , haven't thought about pipefail . taking my words back :)

Reply:
Frankly speaking, I would remove those `cat`'s and `recording_info`'s completely. Maybe leave a `record_info` about what we are downloading.